src/idl_gen_general.cpp

```diff
@@ -968,8 +968,13 @@ class GeneralGenerator : public BaseGenerator {
               if (lang_.language != IDLOptions::kCSharp)
                 code += type_name + " obj, ";
               getter = obj + ".__assign";
+            } else if (vectortype.base_type == BASE_TYPE_UNION) {
+              if (lang_.language != IDLOptions::kCSharp)
```

Comment:
I know this is additional testing burden, but might it be worth it generating this code for C# as well while you're at it? that be cleaner than all these language conditionals. There's a test script that runs with mono on linux/mac, or a test project for VS.

Reply:
Hmm, I can try doing it on Linux and see what happens.

Reply:
Thanks :)